### ISSUE TITLE
Performance reduction of VQE with UCCSD ansatz in Qiskit 0.13.0

### ISSUE THREAD

**Issue by mmetcalf14** [Qiskit/qiskit-metapackage#665]
### Informations

- **Qiskit version**: 0.10.0-0.13.0
- **Python version**: 3.7
- **Operating system**: Mac OS Catalina

### What is the current behavior?

We have been using Qiskit chemistry to run VQE simulations of a CAS(10,10) molecule. The active space of this molecule is encoded on 20 qubits with a ~200 param UCCD ansatz. We started running these simulations using the COBYLA optimizer back in May with Qiskit 0.10.0. Run time for a VQE calculation took ~ 30 hours. After a hiatus, we resumed testing this molecule with Qiskit 0.12.0 and the run time had increased to ~60 hours. Recently, we ran this calculation again with Qiskit 0.13.0 and run time was ~7 days! Digging into the algorithm, I determined most of the time is spent in circuit construction. 

### Steps to reproduce the problem

Use the driver to simulate a molecule with 10 spatial orbitals and 10 electrons using VQE. Truncate the UCCSD ansatz to only include 200 double excitations/variational parameters. Output the run time for a single optimization iteration and multiply by 1500 (max_evals in my calc) using Qiskit 0.13.0. Repeat with previous versions of Qiskit.  

### What is the expected behavior?

Consistent simulation times with each version of Qiskit

### Suggested solutions
Rather than reconstructing the circuit for each energy evaluation in var_form.construct_circuit(), provide a way to update the rotation angles to the optimizer's new parameter values in the variational circuit.

**Comment by dongreenberg** [comment]
Hi Mekena,

Thanks for bringing this up. We're working on this issue. A few quick questions to see whether you can make some immediate gains:
- Are you using the statevector_simulator or qasm_simulator, and BasicAer or Aer?
- Are you using a `QuantumInstance` or running via the `run_algorithm` interface? Do you have json validation disabled?

**Comment by mmetcalf14** [comment]
Thank you Don for your expedient reply.

1. I am using the statevector_simulator.
2. After establishing a quantum instance with 'backend = statevector_simulator' and the VQE algorithm, I call the function 
`quantum_instance = QuantumInstance(backend=statevector_simulator)`

`algorithm = VQE(qop_paulis, var_op, optimizer, 'paulis', initial_point=initial_point) `
 
`result = algorithm.run(quantum_instance=quantum_instance)`

I made this update to my code with the new version.
3. I haven't altered the json validation, and it is set to the Qiskit default.
4. Just to make sure this isn't an issue with Catalina, I ran the code on my Ubuntu workstation. I output the time it takes to run through the vqe.energy_evaluation() function for each iteration (~340 s/iteration) and estimated a 7 day completion time. Roughly 2/3 of the time spent doing a VQE iteration is in this function.
